market-factory/src/lib.rs

```diff
@@ -24,26 +31,54 @@ impl Default for MarketFactory {
 #[near_bindgen]
 impl MarketFactory {
     #[init]
-    pub fn new() -> Self {
+    pub fn new(dao_account_id: AccountId) -> Self {
         if env::state_exists() {
             env::panic_str("ERR_ALREADY_INITIALIZED");
         }
 
         Self {
-            markets: UnorderedMap::new(b"s".to_vec()),
+            markets: LookupMap::new(b"s".to_vec()),
+            dao_account_id,
         }
     }
 
     pub fn create_market(&mut self, market_options: Vec<String>) -> String {
         let market_id = self.get_random_market_id();
 
         self.markets.insert(&market_id, &market_options);
+        self.process_market_options(&market_id, market_options);
 
         return market_id;
     }
-}
 
-impl MarketFactory {
+    fn process_market_options(&self, market_id: &String, market_options: Vec<String>) -> bool {
+        for market_option in market_options {
+            self.create_proposal(&market_id, market_option);
+        }
+
+        return true;
+    }
+
+    fn create_proposal(&self, market_id: &String, market_option: String) -> bool {
+        let dao_proposal_promise = Promise::new(self.dao_account_id.clone()).function_call(
+            "create_proposal".to_string(),
+            json!({ "title": market_option }).to_string().into_bytes(),
+            0,
+            GAS_FOR_CREATE_DAO_PROPOSAL,
+        );
```

Comment:
@susguzman we should create a Sputnik2 DAO proposal here of type `Function Call`. This is a Cross-Contract call promise draft, but it's the general idea.